### PR TITLE
Slideshow feature

### PR DESCRIPTION
**Pull request by tetondev**
Fixing issue where audio `.play()` was called before the audio had time to render. Using the initialized audio files as the slide audio, rather than re-initializing the same audio file.

### CODE REVIEW

File: fs-album-slideshow.html [hunk 1]
```diff
@@ -1142,6 +1150,21 @@ <h4>[[i18n('slideshow_controls_slideshow_speed')]]</h4>
     blurButton: function(e) {
       // If a keypress was used e.currentTarget is 'document' which has no 'blur'
       if (e.currentTarget.blur) e.currentTarget.blur(e);
+    },
+    showAudioError: function() {
+      this.showAudioError = true;
+    },
+    getAudioPlayer: function() {
+      if (!this.audioTracks || !this.audioTracks.length) return false;
+      if (!this.audioTracks[this.currentSlideIndex]) debugger
```
Comment (jasonallen68) [inline fs-album-slideshow.html]: Should probably remove the `debugger` statement.

File: fs-album-slideshow.html [hunk 2]
```diff
@@ -1068,6 +1075,7 @@ <h4>[[i18n('slideshow_controls_slideshow_speed')]]</h4>
      */
     shouldShowPhaudioPlayer: function(currentSlideIndex, artifactsLength) {
       if (!artifactsLength) return false;
+      if (!this.artifacts[currentSlideIndex]) debugger
```
Comment (jasonallen68) [inline fs-album-slideshow.html]: Should probably remove the debugger statement.

File: fs-album-slideshow.html [hunk 3]
```diff
@@ -528,6 +527,10 @@ <h4>[[i18n('slideshow_controls_slideshow_speed')]]</h4>
         type: Array,
         value: function() { return []; }
       },
+      audioTracks: {
+        type: Array,
+        value: function() { return []; }
```
Comment (dcravenus) [inline fs-album-slideshow.html]: Polymer is weird. We have to set the value to a function that returns what we want the initial value to be?

File: bower.json [hunk 4]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "fs-album-display",
-  "version": "4.6.8",
+  "version": "4.6.9",
   "authors": [
     "Josh Crowther <jshcrowthe@gmail.com>"
```
Comment (dcravenus) [inline bower.json]: lol

File: fs-album-slideshow.html [hunk 5]
```diff
@@ -656,29 +660,29 @@ <h4>[[i18n('slideshow_controls_slideshow_speed')]]</h4>
         // Pause the slideshow too
         this.stopSlideshow = true;
         clearTimeout(this.slideDisplayTimeout);
-        if (this.$.audioPlayer.duration) {
-          this.isAudioPlaying = false;
-          this.$.audioPlayer.pause();
-        }
+        this.pauseCurrentAudio(false);
```
Comment (dcravenus) [inline fs-album-slideshow.html]: Do you ever call `pauseCurrentAudio` with true? Also why so explicit with the false here but not in other places?

Comment (dcravenus) [reply fs-album-slideshow.html]: nvm didn't finish reading. you good.

File: fs-album-slideshow.html (hunk 3 above)
Comment (tetondev) [reply fs-album-slideshow.html]: I don't know that the function -> return value is needed, but we've been doing that in a lot of other places and I was following suit.

**Pull request merged**